resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/KubernetesUtils.scala

```diff
@@ -209,4 +215,70 @@ private[spark] object KubernetesUtils extends Logging {
     Hex.encodeHexString(random) + time
   }
 
+  /**
+   * Upload files and modify their uris
+   */
+  def uploadAndTransformFileUris(fileUris: Iterable[String], conf: Option[SparkConf] = None)
+    : Iterable[String] = {
+    fileUris.map { uri =>
+      uploadFileUri(uri, conf)
+    }
+  }
+
+  private def isLocalDependency(uri: URI): Boolean = {
+    Option(uri.getScheme).getOrElse("file") == "file"
+  }
+
+  def renameMainAppResource(resource: String, conf: SparkConf): String = {
+    val resolvedUri = Utils.resolveURI(resource)
```

Comment:
Is it OK to resolve SparkLauncher.NO_RESOURCE? this could be inlined to make sure

Reply:
Yes it should be returned as is will double check